source/MaterialXTest/ShaderValid.cpp

```diff
@@ -385,209 +430,324 @@ static mx::OslValidatorPtr createOSLValidator(bool& orthographicView, std::ostre
 //
 static void runGLSLValidation(const std::string& shaderName, mx::TypedElementPtr element, mx::GlslValidator& validator,
                               mx::GlslShaderGenerator& shaderGenerator, const mx::HwLightHandlerPtr lightHandler, mx::DocumentPtr doc,
-                              std::ostream& log, bool outputMtlxDoc=true, const std::string& outputPath=".")
+                              std::ostream& log, ShaderValid_TestOptions testOptions, bool outputMtlxDoc=true, const std::string& outputPath=".")
 {
-    mx::GenOptions options;
+    std::vector<mx::GenOptions> optionsList;
+    getGenerationOptions(optionsList, testOptions);
 
     if(element && doc)
     {
         log << "------------ Run validation with element: " << element->getNamePath() << "-------------------" << std::endl;
 
-        std::string shaderPath;
-        // Note: mkdir will fail if the directory already exists which is ok.
-        mx::makeDirectory(outputPath);
-        shaderPath = mx::FilePath(outputPath) / mx::FilePath(shaderName);
-
-        mx::ShaderPtr shader;
-        try
-        {
-			options.hwTransparency = mx::isTransparentSurface(element, shaderGenerator);
-            shader = shaderGenerator.generate(shaderName, element, options);
-        }
-        catch(mx::ExceptionShaderGenError e)
-        {
-            log << ">> " << e.what() << "\n";
-            shader = nullptr;
-        }
-        CHECK(shader != nullptr);
-        if (shader == nullptr)
+        for (auto options : optionsList)
         {
-            log << ">> Failed to generate shader\n";
-            return;
-        }
-        CHECK(shader->getSourceCode(mx::HwShader::PIXEL_STAGE).length() > 0);
-        CHECK(shader->getSourceCode(mx::HwShader::VERTEX_STAGE).length() > 0);
-
-        if (outputMtlxDoc)
-        {
-            mx::writeToXmlFile(doc, shaderPath + ".mtlx");
-        }
+            std::string shaderPath;
+            mx::FilePath outputFilePath = outputPath;
+            // Use separate directory for reduced output
+            if (options.shaderInterfaceType == mx::SHADER_INTERFACE_REDUCED)
+            {
+                outputFilePath = outputFilePath / mx::FilePath("/reduced");
```

Comment:
To preserve the current behaviour of testing which is reliant on files names, we just create a new "reduced" sub-directory and run the same testing there.